api/webhook/post.go

```diff
@@ -273,399 +272,52 @@ func PostWebhook(c *gin.Context) {
-
-				util.HandleError(c, http.StatusBadRequest, retErr)
-
-				h.SetStatus(constants.StatusFailure)
-				h.SetError(retErr.Error())
-
-				return
-			}
-		}
-
-		b.SetPipelineID(pipeline.GetID())
-
-		// create the objects from the pipeline in the database
-		// TODO:
-		// - if a build gets created and something else fails midway,
-		//   the next loop will attempt to create the same build,
-		//   using the same Number and thus create a constraint
-		//   conflict; consider deleting the partially created
-		//   build object in the database
-		err = build.PlanBuild(ctx, database.FromContext(c), p, b, repo)
-		if err != nil {
-			retErr := fmt.Errorf("%s: %w", baseErr, err)
-
-			// check if the retry limit has been exceeded
-			if i < retryLimit-1 {
-				logrus.WithError(retErr).Warningf("retrying #%d", i+1)
-
-				// reset fields set by cleanBuild for retry
-				b.SetError("")
-				b.SetStatus(constants.StatusPending)
-				b.SetFinished(0)
-
-				// continue to the next iteration of the loop
-				continue
-			}
-
-			util.HandleError(c, http.StatusInternalServerError, retErr)
-
-			h.SetStatus(constants.StatusFailure)
-			h.SetError(retErr.Error())
-
-			return
-		}
+	// capture the build, repo, and user from the items
+	b, repo, u := item.Build, item.Repo, item.User
 
-		// break the loop because everything was successful
-		break
-	} // end of retry loop
+	// set hook build_id to the generated build id
+	h.SetBuildID(b.GetID())
 
-	// send API call to update repo for ensuring counter is incremented
-	repo, err = database.FromContext(c).UpdateRepo(ctx, repo)
 	if err != nil {
-		retErr := fmt.Errorf("%s: failed to update repo %s: %w", baseErr, repo.GetFullName(), err)
-		util.HandleError(c, http.StatusBadRequest, retErr)
-
-		h.SetStatus(constants.StatusFailure)
-		h.SetError(retErr.Error())
-
-		return
-	}
-
-	// return error if pipeline didn't get populated
-	if p == nil {
-		retErr := fmt.Errorf("%s: failed to set pipeline for %s: %w", baseErr, repo.GetFullName(), err)
-		util.HandleError(c, http.StatusBadRequest, retErr)
-
-		h.SetStatus(constants.StatusFailure)
-		h.SetError(retErr.Error())
-
-		return
-	}
-
-	// return error if build didn't get populated
-	if b == nil {
-		retErr := fmt.Errorf("%s: failed to set build for %s: %w", baseErr, repo.GetFullName(), err)
-		util.HandleError(c, http.StatusBadRequest, retErr)
-
 		h.SetStatus(constants.StatusFailure)
-		h.SetError(retErr.Error())
+		h.SetError(err.Error())
 
 		return
 	}
 
-	// send API call to capture the triggered build
-	b, err = database.FromContext(c).GetBuildForRepo(ctx, repo, b.GetNumber())
-	if err != nil {
-		retErr := fmt.Errorf("%s: failed to get new build %s/%d: %w", baseErr, repo.GetFullName(), b.GetNumber(), err)
-		util.HandleError(c, http.StatusInternalServerError, retErr)
-
-		h.SetStatus(constants.StatusFailure)
-		h.SetError(retErr.Error())
+	if !pushed {
+		// if the build was not pushed to the queue, it was skipped
+		h.SetStatus(constants.StatusSkipped)
```

Comment:
you would reach this when `err` is nil and `pushed` is false which looks to be possible in two instances:
1. build got skipped, or
2. unable to set route for build (https://github.com/go-vela/server/pull/1063/files#diff-0e4342c1b8362db4ef095686b38347fc9fada69877f4432cdcd26a99b9c379feR434)

i'm guessing we probably want to update 2. to actually return an error?

Reply:
Ah, yep, you're right. 2 should be an error like it is today